src/tests/unit/platform/test_users.py

```diff
@@ -0,0 +1,217 @@
+    assert len(list_users) == 5
+    assert list_users[0].last_name == 'Mercer'
+    assert list_users[1].login_name == 'mreynolds@browncoats.org'
+    assert list_users[2].email == 'djackson@stargate-command.af.mil'
+    assert list_users[3].phone == '714-555-4567'
+    assert list_users[4].first_name == 'Beckett'
+
+
+def test_get_and_modify_user(cbcsdk_mock):
+    """Tests retrieving and modifying a user."""
+    def check_put(url, body, **kwargs):
+        assert body['login_id'] == 6942
+        assert body['login_name'] == 'jsheridan@babylon5.com'
+        assert body['email'] == 'jsheridan@zhadum.net'
+        return None
+
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    cbcsdk_mock.mock_request('PUT', '/appservices/v6/orgs/test/users/6942', check_put)
+    api = cbcsdk_mock.api
+    user = api.select(User, 6942)
+    assert user.last_name == 'Sheridan'
+    assert user.first_name == 'John'
+    assert user.login_name == 'jsheridan@babylon5.com'
+    assert user.email == 'jsheridan@babylon5.com'
+    assert user.urn == 'psc:user:test:6942'
+    user.email = 'jsheridan@zhadum.net'
+    user.save()
+    assert user.last_name == 'Sheridan'
+    assert user.first_name == 'John'
+    assert user.login_name == 'jsheridan@babylon5.com'
+    assert user.email == 'jsheridan@zhadum.net'
+
+
+def test_get_and_delete_user(cbcsdk_mock):
+    """Tests retrieving and deleting a user."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/users/3978', None)
+    api = cbcsdk_mock.api
+    user = api.select(User, 3978)
+    assert user.last_name == 'Mariner'
+    assert user.first_name == 'Beckett'
+    assert user.login_name == 'bmariner@cerritos.starfleet.mil'
+    assert user.email == 'bmariner@cerritos.starfleet.mil'
+    assert user.urn == 'psc:user:test:3978'
+    user.delete()
+
+
+def test_get_user_and_reset_googleauth(cbcsdk_mock):
+    """Tests retrieving a user and resetting Google authentication."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/users/4338/googleauth', None)
+    api = cbcsdk_mock.api
+    user = api.select(User, 4338)
+    assert user.last_name == 'Jackson'
+    assert user.first_name == 'Daniel'
+    assert user.login_name == 'djackson@stargate-command.af.mil'
+    assert user.email == 'djackson@stargate-command.af.mil'
+    assert user.urn == 'psc:user:test:4338'
+    user.reset_google_authenticator_registration()
+
+
+def test_get_nonexistent_user(cbcsdk_mock):
+    """Test retrieving a nonexistent user."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    api = cbcsdk_mock.api
+    with pytest.raises(ObjectNotFoundError):
+        api.select(User, 999)
+
+
+def test_attempt_to_create_naively(cb):
```

Comment:
Is this a behavior we want to prevent or should we support this by checking if an ID exists and if it doesn't then create a new User?

Reply:
We don't support it for Grant; we require that they go through the Builder or use a template. This is by analogy.

Reply:
Then let's change the test name to be `test_unsupported_create_by_update` as the language here is not appropriate in my opinion

Reply:
As you wish.